### DISCUSSION TITLE
Checkout of a bitbucket.org fails

### DISCUSSION THREAD

**Discussion by vadi2** [community/community#26162] [Actions]
<p>Using v2 checkout action, the checkout of a submodule hosted on <a href="http://bitbucket.com" rel="nofollow noopener">bitbucket.com</a> consistently fails:</p>
<pre><code>Run # If your submodules are configured to use SSH instead of HTTPS please uncomment the following line
Submodule '3rdparty/dblsqd' (https://bitbucket.org/pentacent/dblsqd-sdk-qt.git) registered for path '3rdparty/dblsqd'
Submodule '3rdparty/edbee-lib' (https://github.com/Mudlet/edbee-lib.git) registered for path '3rdparty/edbee-lib'
Submodule '3rdparty/lua_code_formatter' (https://github.com/martin-eden/lua_code_formatter.git) registered for path '3rdparty/lcf'
Submodule '3rdparty/qtkeychain' (https://github.com/Mudlet/qtkeychain.git) registered for path '3rdparty/qtkeychain'
Submodule '3rdparty/sparkle-glue' (https://github.com/Mudlet/mixing-cocoa-and-qt.git) registered for path '3rdparty/sparkle-glue'
Cloning into '/home/runner/work/Mudlet/Mudlet/3rdparty/dblsqd'...
fatal: could not read Username for 'https://bitbucket.org': No such device or address
fatal: clone of 'https://bitbucket.org/pentacent/dblsqd-sdk-qt.git' into submodule path '/home/runner/work/Mudlet/Mudlet/3rdparty/dblsqd' failed
Failed to clone '3rdparty/dblsqd'. Retry scheduled
Cloning into '/home/runner/work/Mudlet/Mudlet/3rdparty/edbee-lib'...
Cloning into '/home/runner/work/Mudlet/Mudlet/3rdparty/lcf'...
Cloning into '/home/runner/work/Mudlet/Mudlet/3rdparty/qtkeychain'...
Cloning into '/home/runner/work/Mudlet/Mudlet/3rdparty/sparkle-glue'...
Cloning into '/home/runner/work/Mudlet/Mudlet/3rdparty/dblsqd'...
fatal: could not read Username for 'https://bitbucket.org': No such device or address
fatal: clone of 'https://bitbucket.org/pentacent/dblsqd-sdk-qt.git' into submodule path '/home/runner/work/Mudlet/Mudlet/3rdparty/dblsqd' failed
Failed to clone '3rdparty/dblsqd' a second time, aborting
##[error]Process completed with exit code 1.
</code></pre>
<p>This is the run in question: <a href="https://github.com/Mudlet/Mudlet/runs/496925890" rel="nofollow noopener">https://github.com/Mudlet/Mudlet/runs/496925890</a></p>

**Comment by vadi2** [comment]
<p>As a workaround, checkout v1 does the job.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/vadi2">@vadi2</a> ,</p>
<p>I also tested the same checkout submodules steps mentioned on the docs of the checkout action, I also get the same error when trying to checkout the submodules from Bitbucket repositories. But if the submodules is from the GitHub repositories, the same steps can work fine.</p>
<p>Not sure if we need to provide more other authorizations when checkout submodules from other services when we use these checkout submodules steps.</p>
<p>I have reported this ticket to the appropriate engineering team, the team will evaluate and investigate this question. If they have any update, I will notify you in time, and sometimes the appropriate engineers may will directly reply you here.</p>
<p>In addition, you also can try to change the checkout submodules steps as below:</p>
<pre><code>- name: Checkout
        uses: actions/checkout@v2
      - name: Checkout submodules
        shell: bash
        run: |
          git submodule init
          git submodule update
</code></pre>
<p>I have tested this, it can checkout the submodules from Bitbucket without any error. More details, you can reference the docs about <a href="https://git-scm.com/book/en/v2/Git-Tools-Submodules" rel="nofollow noopener"><strong>Git - Submodules</strong></a>.</p>

**Comment by ericsciple** [comment]
<p>Support for submodules was recently added in in master. Still collecting feedback before updating <code>v2</code> tag. Curious if it works with <code>actions/checkout@master</code> with the new <code>submodules</code> input <a href="https://github.com/actions/checkout/blob/master/README.md" rel="nofollow noopener">https://github.com/actions/checkout/blob/master/README.md</a> ?</p>
